qemu/tests/edk2_check_mor.py

```diff
@@ -0,0 +1,82 @@
+import re
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_package
+from avocado.utils import process
+from avocado.utils.path import find_command
+from avocado.utils.path import CmdNotFoundError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify MOR enabled in edk2 build
+
+    1. Boot guest under secure mode and check if the guest is signed
+    2. Check if secure boot is enabled inside guest
+    3. Reboot and shutdown the guest
+    4. Check MOR message after shutdown the guest
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def _check_signed():
+        """ Check and return if guest is signed """
+        return True if re.search(sign_keyword, sign_info) else False
+
+    package = params["package_installed"]
+    install_status = utils_package.package_install(package)
+    if not install_status:
+        test.error(f"Failed to install {package}.")
+    try:
+        find_command(params["cmd_installed"])
+    except CmdNotFoundError as e:
+        test.error(e.__str__())
+    params['ovmf_vars_filename'] = 'OVMF_VARS.secboot.fd'
+    env_process.preprocess_vm(test, params, env, params['main_vm'])
+    vm = env.get_vm(params['main_vm'])
+    vm.create(params=params)
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    check_sign_cmd = params['check_sign_cmd']
+    sign_keyword = params['sign_keyword']
+    if session.cmd_status('which pesign') != 0:
+        install_status = utils_package.package_install('pesign', session)
+    if not install_status:
+        test.error("Failed to install pesign.")
+    vmlinuz = '/boot/vmlinuz-%s' % session.cmd_output('uname -r')
+    check_sign_cmd %= vmlinuz
+    sign_info = session.cmd_output(check_sign_cmd)
+    signed = _check_signed()
+    error_context.context('Guest signed status is %s' % signed, test.log.info)
+    check_cmd = params['check_secure_boot_enabled_cmd']
+    status, output = session.cmd_status_output(check_cmd)
+    if status:
+        test.cancel('Secure boot is not enabled,'
+                    'MOR must run under secure mode')
+    if not signed:
```

Comment:
move the if under line 53?

Reply:
updated, check whether the guest has been signed after checking secure boot enabled. If the guest is not signed, it should not boot succeed under secure mode. Thanks.